src/build.ts

```diff
@@ -175,7 +175,7 @@ export async function writeTypes(nitro: Nitro) {
       ],
     };
     await writeFile(
-      join(nitro.options.buildDir, "types/tsconfig.json"),
+      join(nitro.options.buildDir, nitro.options.typescript.tsconfigPath),
```

Comment:
I think we should switch to `resolve` now to allow absolute paths wdyt?

```suggestion
      resolve(nitro.options.buildDir, nitro.options.typescript.tsconfigPath),
```